nevergrad/optimization/optimizerlib.py

```diff
@@ -1823,3 +1823,82 @@ def _internal_tell_not_asked(self, candidate: p.Parameter, value: float) -> None
         raise base.TellNotAskedNotSupportedError
 
 
+@registry.register
+class NGOpt(base.Optimizer):
+    """Nevergrad optimizer by competence map. You might modify this one for designing youe own competence map."""
+
+    def __init__(self, parametrization: IntOrParameter, budget: Optional[int] = None, num_workers: int = 1) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        assert budget is not None
+        descr = self.parametrization.descriptors
+        self.has_noise = not (descr.deterministic and descr.deterministic_function)
+        self.fully_continuous = descr.continuous
+        all_params = paramhelpers.flatten_parameter(self.parametrization)
+        self.has_discrete_not_softmax = any(isinstance(x, p.BaseChoice) for x in all_params.values())
+        arity: int = max(len(param.choices) if isinstance(param, p.BaseChoice) else -1 for param in all_params.values())
+        if self.has_noise and (self.has_discrete_not_softmax or not self.parametrization.descriptors.metrizable):
+            self.optim: base.Optimizer = RecombiningPortfolioOptimisticNoisyDiscreteOnePlusOne(self.parametrization, budget, num_workers)
```

Comment:
to simplify the code, you can keep track of the selected `OptimClass`
and then you only instantiate an the one `OptimClass(self.parametrization, budget, num_workers)`
while that wont remove likes, it will remove a lot of of copied code